columnflow/production/cms/muon.py

```diff
@@ -18,15 +18,15 @@
     uses={
         "Muon.pt", "Muon.eta",
     },
-    produces={
-        "muon_weight", "muon_weight_up", "muon_weight_down",
-    },
+    # produces in the init
     # only run on mc
     mc_only=True,
     # function to determine the correction file
     get_muon_file=(lambda self, external_files: external_files.muon_sf),
     # function to determine the muon weight config
     get_muon_config=(lambda self: self.config_inst.x.muon_sf_names),
+    weight_name=(lambda self: "muon_weight"),
```

Comment:
Why is this a function? Is it not enough to just make it a simple string? This adds flexibility for sure, but the name `weight_name` suggests the name itself, not a function that generates the name. Is there a specific use case in mind?

Reply:
I agree, I think leaving the name fixed is more resilient to other things that might happen in the workflow. Thanks!